### PR TITLE
Add AB test handlers for nlp weighting

### PR DESCRIPTION
**Pull request by lindenmckenzie**
### What

Added AB testing functionality for NLP weighting. 

### How to review

Check tests pass, code looks good. 

Optionally:
- port forward to api router
- set ab_test cookie to appropriate value
- see the requests coming in to search-api via kibana with nlp_weighting=true vs false
- check the gtm data layer script to see different values being assigned to "abTest"

### Who can review

Frontend Go dev.

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -355,6 +362,12 @@ func read(w http.ResponseWriter, req *http.Request, cfg *config.Config, zc Zebed
 		return
 	}
 
+	validatedQueryParams.NLPWeightingEnabled = nlpWeightingEnabled
+
```
Comment (mr-nick17) [inline handlers/handlers.go:366]: The log is related to the function so I think there should be no whitespace

File: handlers/handlers_test.go [hunk 2]
```diff
@@ -180,14 +180,30 @@ func TestUnitReadSuccess(t *testing.T) {
...
+			})
+		})
+
+		Convey("When read is called with NLP switched on", func() {
+			read(w, req, cfg, mockedZebedeeClient, mockedRendererClient, mockedSearchClient, accessToken, collectionID, englishLang, *mockCacheList, "search", true)
+
+			Convey("Then a 200 OK status should be returned", func() {
+				So(w.Code, ShouldEqual, http.StatusOK)
+
+				So(len(mockedRendererClient.BuildPageCalls()), ShouldEqual, 1)
```
Comment (mr-nick17) [inline handlers/handlers_test.go:202]:
```suggestion
				So(mockedRendererClient.BuildPageCalls(), ShouldHaveLength, 1)
```

File: handlers/handlers_test.go [hunk 3]
```diff
@@ -180,14 +180,30 @@ func TestUnitReadSuccess(t *testing.T) {
...
+				So(len(mockedSearchClient.GetSearchCalls()), ShouldEqual, 2)
```
Comment (mr-nick17) [inline handlers/handlers_test.go:203]: ..and here

File: handlers/handlers_test.go [hunk 4]
```diff
@@ -180,14 +180,30 @@ func TestUnitReadSuccess(t *testing.T) {
...
+				So(len(mockedZebedeeClient.GetHomepageContentCalls()), ShouldEqual, 1)
```
Comment (mr-nick17) [inline handlers/handlers_test.go:204]: and here

File: handlers/handlers_test.go [hunk 5]
```diff
@@ -239,7 +255,7 @@ func TestUnitReadFailure(t *testing.T) {
 		So(err, ShouldBeNil)
 
 		Convey("When read is called", func() {
-			read(w, req, cfg, mockedZebedeeClient, mockedRendererClient, mockedSearchClient, accessToken, collectionID, englishLang, *mockCacheList, "search")
+			read(w, req, cfg, mockedZebedeeClient, mockedRendererClient, mockedSearchClient, accessToken, collectionID, englishLang, *mockCacheList, "search", false)
```
Comment (mr-nick17) [inline handlers/handlers_test.go:258]: Not for here (maybe) but I think these function signatures for all the handlers are getting quite big and are all quite similar. Do you think they should be a method with a receiver type?

**Review by mr-nick17 — COMMENTED** [review]
Some minor comments/suggestions

File: handlers/handlers_test.go (hunk 5 above)
Comment (lindenmckenzie) [reply handlers/handlers_test.go:258]: Take the point but later :)

**Review by mr-nick17 — APPROVED** [review]

**Pull request merged**